### DISCUSSION TITLE
Auditing repo

### DISCUSSION THREAD

**Discussion by ahozdemir** [trimble-oss/dba-dash#662] [Ideas]
I think, It will be great to custom auditing to store audit data on  repo, It seems to easy to develop, may be we  would add another project's source 
one of those project example https://github.com/iraklisk/SqlAuditor
https://github.com/iraklisk/SqlAuditor

**Comment by DavidWiseman** [comment]
It looks like that project is build around SQL Trace.  DBA Dash can capture some of the same data already using extended events. DBA Dash captures Batch Completed & RPC Completed events when you enable slow query capture.  By default this is queries that take longer than 1 second to run but there is the option to customize this.  If you need more advanced filtering you can choose to persist the extended events session which will allow you to apply any further customizations to it.  It's just designed to process Batch/RPC Completed events.

**Comment by ahozdemir** [comment]
Actually, it could be done with manually, with sql auditing, But it will be great what if an option to add'ing auditing standart for concern IT's like

![image](https://github.com/trimble-oss/dba-dash/assets/6367946/e89b9a01-73cb-4cce-a41a-625ea2fc97ec)

and also It will be able to record those records centrally and any case of change will belogged

for example just logged when whom and executed command to object like

10.1.2023 22:00:00  user1   select/insert/update/delete     table/sp/view (name)


regards

**Comment by ahozdemir** [comment]
Hi, I am big fan of your project by the way, great job, congs...
Is there any way to add auditing to  as I wrote before like those format;
Exaple:
10.1.2023 22:00:00 user1 select/insert/update/delete table/sp/view (name)  

even though Extended events captured data added to central repo in dba dash

**Comment by DavidWiseman** [comment]
Hi, thanks for the suggestion.  This feature isn't something I would make use out of at the moment so I'm unlikely to develop it.  Even if implemented, the volume of data from an audit could be a concern.
